src/pages/EditMonitor.vue

```diff
@@ -1265,7 +1265,7 @@ message HealthCheckResponse {
                 this.monitor.body = JSON.stringify(JSON.parse(this.monitor.body), null, 4);
             }
 
-            if (this.monitor.type && this.monitor.type !== "http" && (this.monitor.type !== "keyword" || this.monitor.type !== "json-query")) {
+            if (this.monitor.type && this.monitor.type !== "http" && this.monitor.type !== "keyword" && this.monitor.type !== "json-query") {
```

Comment:
Now I get what this code should do.
The following does communicate that intend more clearly imo, what do you think?

```suggestion
            const monitorTypesWithEncodingAllowed = ["http", "keyword", "json-query"];
            if (!monitorTypesWithEncodingAllowed.includes(this.monitor.type)) {
```

Reply:
That looks like a good idea - I'm not familiar with this code, so just kept it as it was, but I like that. Easy for anyone to see and understand.

Reply:
@CommanderStorm do you wanna commit it to the PR or should I do it?

Reply:
Only you (author) or maintainers can apply commits => please do so if you think that this is better.

Here is the verification that it is identical:
![image](https://github.com/louislam/uptime-kuma/assets/26258709/90b1691b-2a06-4949-b398-603c1be5ab9c)